hotel-management/src/hooks/room/useFetchRooms.ts

```diff
@@ -1,41 +1,43 @@
 import { useEffect, useState } from 'react';
 
-import { API_ENDPOINT } from '@/constants';
+import { API_ENDPOINT, ROOM_MESSAGE } from '@/constants';
 import HttpService from '@/services/httpService';
-import { Room, ApiResponse } from '@/types';
+import { Room, ApiResponse, RoomParams } from '@/types';
 
 export function useFetchRooms() {
   const [rooms, setRooms] = useState<Room[]>([]);
   const [error, setError] = useState<string | null>(null);
-  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
   const { ROOMS } = API_ENDPOINT;
+  const { SET_ROOMS_FAILED, UNKNOWN_ERROR } = ROOM_MESSAGE;
 
-  useEffect(() => {
-    (async () => {
-      setIsLoading(true);
-      setError(null);
+  const fetchRooms = async (params?: RoomParams) => {
+    setError(null);
```

Comment:
Why we need to set error null here, does error default is null?

Reply:
because `setError(null)` before starting the fetch will make sure that any previous errors are cleared out.